esp/templates/program/modules/resourcemodule/classrooms.html

```diff
@@ -74,3 +97,71 @@ <h2>Step 3: Classrooms</h2>
 {% endifequal %}
 
 <br />
+
+<!--This branch has a fix for when the form is originally hidden-->
+<script src="https://rawgit.com/elo80ka/django-dynamic-formset/f5f68c818953ddf43ffa7cb2e9aa391c3d4dfb47/src/jquery.formset.js"></script>
+<!--jQuery.initialize plugin is created to help maintain dynamically created elements on the page-->
+<script src="https://rawgit.com/pie6k/jquery.initialize/master/jquery.initialize.js"></script>
+<script>
+<!--Formset settings-->
+$j('.furnishing-formset').formset({
+    prefix: 'furnishings',
+    formTemplate: '#id_empty_form',
+    addText: 'add furnishing',
+    deleteText: 'remove',
+    addCssClass: 'furnishing-add',
+    deleteCssClass: 'furnishing-delete',
+    formCssClass: 'furnishing-dynamic-form',
+});
+
+<!--Get choices for a furnishing via POST-->
+function getChoices(furnishing, callback){
+    var data = {furnishing: furnishing, csrfmiddlewaretoken: csrf_token()};
+    $j.post('/manage/{{ prog.url }}/ajaxfurnishingchoices', data, "json").success(callback);
+}
+
+<!--Update the option field to reflect whether there are specific choices (dropdown) or not (open response). If a value is supplied, set that value as selected afterwards.-->
+function update_choices(obj, value) {
+    var furnishing = $j(obj).val()
+    getChoices(furnishing, function(response) {
+        var choices = response.choices;
+        var num = $j(obj).attr('id').split("-")[1];
+        $j('#id_furnishings-' + num + '-choice').remove();
+        if (choices.length > 1) {
+            $j(obj).after('<select style="margin-left: 4px" id="id_furnishings-' + num + '-choice" name="furnishings-' + num + '-choice" type="text"></select>');
+            $j('#id_furnishings-' + num + '-choice').append('<option>(option)</option>');
+            for (i in choices) {
+                $j('#id_furnishings-' + num + '-choice').append('<option value=' + choices[i] + '>' + choices[i] + '</option>');
+            }
+        } else {
+            $j(obj).after('<input style="margin-left: 4px" id="id_furnishings-' + num + '-choice" name="furnishings-' + num + '-choice" maxlength="50" placeholder="(option)" type="text"></select>');
+        }
+        if (value) {
+            $j('#id_furnishings-' + num + '-choice').val(value);
+        }
+    });
+}
+
+<!--Upon page load and dynamic creation of new selects-->
+$j.initialize("select", function() {
```

Comment:
Hmm, can you instead do this where you create the select?  It's a bit surprising to see it here, and potentially has unexpected consequences (say, if someone else writes other JS to add another unrelated select to the page in a similar fashion).

Reply:
The selects are made by the dynamic formset plugin, so I can't really move this there. Also, the `target` option below this indicates to `initialize` that it should only look for `select` objects within the specified object (the div with the id `furnishing_formset`), so that should keep the scope limited.